### ISSUE TITLE
Should sidebar be dismissed with <esc> ?

### ISSUE THREAD

**Issue by Carreau** [jupyterlab/jupyterlab#277]
Currently the sidebar with filebrowser looks like permanently  on the side. 
It is useful as it allows to quickly access files/actions.

It is understandable that one want to hide it with a keyboard shortcut, 
If I understand correctly, it is possible with Cmd-ShiftP, or escape. 

I though find myself every now and then pressing `<esc>` in command mode and dismissing the console, which, with the resizing /reflowing of notebook tend to perturb me. As it can already be toggled by `Ctrl-Shift-P`, is it a good idea to unbind `<esc>` ? 

I'm also wondering if the command palette should be floating, to avoid visual notebook reflow/movement.
I'm torn on this one as definitively see the advantage and usefulness of a side panel, but this feel like a floating dialog that is automatically dismissed once you trigger an action is also extremely useful.

One of the possibility I see would be that Cmd-Shift-P have a conditional behavior depending of wether the side panel is visible:

If it's hidden:
- bring a floating version that is dismissed once command executed.
  If it's visible:
- current behavior.

Or something similar.

**Comment by ellisonbg** [comment]
We are working on some redesign of the R/L panels that will address some of these interactions. Closing for now as no action...

**Comment by Carreau** [comment]
I would appreciate if you could link to newer discussions in general, It's becoming extremely hard to follow what is happening if every time I come to an issue I am subscribed to I have to literally look through all of the open and closed issue to know which one is the new one I should follow.

**Comment by ellisonbg** [comment]
Closing in favor of #3393
